File: snippets/python/src/req/fetch.py
# ANCHOR: full
import asyncio
from datetime import timedelta

from nostr_sdk import Client, Filter, Events, Kind, KindStandard, RelayUrl


async def fetch():
    # ANCHOR: client
    client = Client()

    relay_url = RelayUrl.parse("wss://relay.damus.io")
    await client.add_relay(relay_url)
    await client.connect()
    # ANCHOR_END: client

    # ANCHOR: fetch
    filter: Filter = Filter().kind(Kind.from_std(KindStandard.METADATA)).limit(3)
    events: Events = await client.fetch_events(filter, timedelta(seconds=10))
    # ANCHOR_END: fetch

    # ANCHOR: fetch-from
    filter: Filter = Filter().kind(Kind.from_std(KindStandard.TEXT_NOTE)).limit(5)
    events: Events = await client.fetch_events_from([relay_url], filter, timedelta(seconds=10))
    # ANCHOR_END: fetch-from


if __name__ == '__main__':
   asyncio.run(fetch())
# ANCHOR_END: full

File: snippets/rust/src/req/fetch.rs
// ANCHOR: full
use std::time::Duration;

use nostr_sdk::prelude::*;

pub async fn fetch() -> Result<()> {
    // ANCHOR: client
    let client = Client::default();

    client.add_relay("wss://relay.damus.io").await?;
    client.connect().await;
    // ANCHOR_END: client

    // ANCHOR: fetch
    let filter: Filter = Filter::new().kind(Kind::Metadata).limit(3);
    let events: Events = client.fetch_events(filter, Duration::from_secs(10)).await?;
    // ANCHOR_END: fetch

    // ANCHOR: fetch-from
    let filter: Filter = Filter::new().kind(Kind::TextNote).limit(5);
    let events: Events = client
        .fetch_events_from(["wss://relay.damus.io"], filter, Duration::from_secs(10))
        .await?;
    // ANCHOR_END: fetch-from

    Ok(())
}
// ANCHOR_END: full
